File: java/base/src/main/vsdk/toolkit/environment/background/Background.java
package vsdk.toolkit.environment.background;
import java.io.Serial;

import vsdk.toolkit.common.Entity;
import vsdk.toolkit.common.color.ColorRgb;
import vsdk.toolkit.common.linealAlgebra.Vector3Dd;

public abstract class Background extends Entity
{
    @SuppressWarnings("FieldNameHidesFieldInSuperclass")
    @Serial private static final long serialVersionUID = 20150218L;
    
    public Background() {

    }
    public abstract ColorRgb colorInDireccion(Vector3Dd d);
}

File: typescript/base/src/main/vsdk/toolkit/environment/background/Background.ts
import { Entity } from "../../common/Entity.js";
import { ColorRgb } from "../../common/color/ColorRgb.js";
import { Vector3Dd } from "../../common/linealAlgebra/Vector3Dd.js";

/**
 * A model-side source of colour for a viewing direction.
 *
 * Backgrounds deliberately do not depend on a rendering backend.  A concrete
 * background can therefore be sampled by either the software or GPU renderer.
 */
export abstract class Background extends Entity {
    public abstract colorInDireccion(d: Vector3Dd): ColorRgb | null;
}
